File: proddata/data/accounts-merge/C++/Driver.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <stack>
#include <string>
#include <vector>
#include <sstream>
#include <iterator>
#include <unordered_map>
#include <unordered_set>

#include "Solution.hpp"

using namespace std;

void Print(vector<vector<string>> actual_output, vector<vector<string>> expected_output)
{
    cout << "Result: Failed" << endl;
    cout << "Actual Output:" << endl;
    for (int i = 0; i < actual_output.size(); i++)
    {
        for (int j = 0; j < actual_output[i].size(); j++)
        {
            cout << actual_output[i][j] << " ";
        }
        cout << endl;
    }
    cout << endl;

    cout << "Expected Output:" << endl;
    for (int i = 0; i < expected_output.size(); i++)
    {
        for (int j = 0; j < expected_output[i].size(); j++)
        {
            cout << expected_output[i][j] << " ";
        }
        cout << endl;
    }
}

int main()
{
    int vectorSize;
    vector<vector<string>> input;
    vector<vector<string>> expected_output;
    vector<vector<string>> actual_output;
    ifstream infile("../testcases.txt");
    Solution *obj = new Solution();
    string line;
    int status = 0;
    while (getline(infile, line))
    {
        if (line == "input")
        {
            status = 1;
            continue;
        }

        if (line == "output")
        {
            status = 0;
            continue;
        }

        if (line == "check")
        {
            actual_output = obj->accountsMerge(input);

            sort(actual_output.begin(), actual_output.end());

            sort(expected_output.begin(), expected_output.end());

            if (expected_output != actual_output)
            {
                Print(actual_output, expected_output);
                return 0;
            }

            if (expected_output.size() == actual_output.size())
            {
                for (int i = 0; i < expected_output.size(); i++)
                {
                    for (int j = 0; j < expected_output[i].size(); j++)
                    {
                        if (expected_output[i][j] != actual_output[i][j])
                        {
                            Print(actual_output, expected_output);
                            return 0;
                        }
                    }
                }
            }

            input.clear();
            expected_output.clear();
        }

        if (status == 1 && line != "check")
        {
            stringstream ss(line);
            vector<string> new_vec;
            string v;
            while (ss >> v)
            {
                new_vec.push_back(v);
            }
            input.push_back(new_vec);
            new_vec.clear();
        }
        if (status == 0 && line != "check")
        {
            stringstream ss(line);
            vector<string> new_vec;
            string v;
            while (ss >> v)
            {
                new_vec.push_back(v);
            }
            expected_output.push_back(new_vec);
            new_vec.clear();
        }
    }
    cout << "Result: Success" << endl;
    return 0;
}

File: proddata/data/richest_customer_wealth/Python/Driver.py
from xml.etree.ElementPath import xpath_tokenizer
from Solution import Solution
import sys
import os

file1 = open(os.path.join(sys.path[0], '../testcases.txt'), 'r')
Lines = file1.readlines()
input = []
x = []
actual_output = []
expected_output = []
status = 0
obj = Solution()
for line in Lines:
    if (line.strip() == "input"):
        status = 1
        continue
    if (line.strip() == "output"):
        status = 0
        continue
    if (line.strip() == "check"):
        actual_output = obj.maximumWealth(input)
        if (expected_output != actual_output):
            print('Result: Failed')
            print("Input: land= {}".format(x))
            print("Actual Output:", actual_output)
            print("Expected Outuput:", expected_output)
            status = 1
            break
        input.clear()
        x.clear()
    if (status == 1 and line.strip() != "check"):
        new_vec = list(map(int, line.split()))
        input.append(new_vec)
        x.append(new_vec.copy())

    if(status == 0 and line.strip() != "check"):
        expected_output = int(line.strip())


if status == 0:
    print('Result: Success')